File: Data Structures and Algorithms/4. Recursion/Kunal/Arrays/sortedOrNot.cpp
#include <bits/stdc++.h>
using namespace std;

bool sorted_or_not(vector<int>a, int index){
    if(index == a.size() - 1) {
        return true;
    }
    return (a[index] < a[index + 1]) and (sorted_or_not(a, index + 1));
}

void print_array(vector<int>&a) {
    for(int i = 0; i < a.size(); i++) {
        cout << a[i] << " ";
    }
    cout << endl;
}

int main(){
    vector<int> a = {2, 4, 8, 10};
    
    bool check = sorted_or_not(a, 0);
    if(check) {
        cout << "Sorted.\n";
    } else {
        cout << "Not Sorted.\n";
    }
    return 0;
}

File: Data Structures and Algorithms/4. Recursion/Kunal/Arrays/SortedOrNot.java
import java.util.*;
public class SortedOrNot{
    public static boolean sorted(int[] arr, int index) {
        if(index == arr.length - 1)
            return true;
        return arr[index] < arr[index + 1] && sorted(arr, index + 1);
    }

    public static void print_array(int []arr, int n) {
        for(int i = 0; i < n; i++){
            System.out.print(arr[i] + " ");
        }
        System.out.println();
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int []arr = new int[n];
        for(int i = 0; i < n; i++){
            arr[i] = sc.nextInt();
        }
        print_array(arr, n);

        boolean check = sorted(arr, 0);
        System.out.print(check);
    }
}

